XenAdmin/TabPages/CdnUpdates/CdnExpandableRow.cs

```diff
@@ -0,0 +1,248 @@
+                _nameCell.Value as string,
+                string.Empty
+            };
+            details.AddRange(ChildRows.SelectMany(r => r.Export()));
+            return details;
+        }
+    }
+
+
+    internal class PoolUpdateInfoRow : CdnExpandableRow
+    {
+        public PoolUpdateInfoRow(IXenConnection connection, CdnPoolUpdateInfo poolUpdateInfo)
+        {
+            Pool = Helpers.GetPoolOfOne(connection);
+
+            var lastSyncTime = Messages.INDETERMINABLE;
+
+            if (Helpers.XapiEqualOrGreater_23_18_0(connection))
+            {
+                lastSyncTime = Messages.NEVER;
+
+                if (Pool.last_update_sync > Util.GetUnixMinDateTime())
+                {
+                    lastSyncTime = HelpersGUI.DateTimeToString(Pool.last_update_sync.ToLocalTime(), Messages.DATEFORMAT_DMY_HMS, true);
+                }
+            }
+
+            SetValues(Helpers.GetName(connection), Images.GetImage16For(Images.GetIconFor(connection)), lastSyncTime);
+
+            if (poolUpdateInfo == null)
+            {
+                ChildRows = connection.Cache.Hosts
+                    .Select(h => new HostUpdateInfoRow(connection, h, null, null))
+                    .Cast<CdnExpandableRow>().ToList();
+            }
+            else
+            {
+                ChildRows = poolUpdateInfo.HostsWithUpdates
+                    .Select(h => new HostUpdateInfoRow(connection, connection.Resolve(new XenRef<Host>(h.HostOpaqueRef)), poolUpdateInfo, h))
+                    .Cast<CdnExpandableRow>().ToList();
+            }
+        }
+
+        public Pool Pool { get; }
+
+        public override List<CdnExpandableRow> ChildRows { get; } = new List<CdnExpandableRow>();
+    }
+
+
+    internal class HostUpdateInfoRow : CdnExpandableRow
+    {
+        private readonly List<CdnExpandableRow> _childRows = new List<CdnExpandableRow>();
+
+        public HostUpdateInfoRow(IXenConnection connection, Host host, CdnPoolUpdateInfo poolUpdateInfo, CdnHostUpdateInfo hostUpdateInfo)
+        {
+            Connection = connection;
+            Host = host;
+
+            string lastUpdateTime = Messages.NEVER;
+
+            if (Helpers.XapiEqualOrGreater_22_20_0(Host))
+            {
+                var unixMinDateTime = Util.GetUnixMinDateTime();
+                var softwareVersionDate = unixMinDateTime;
+
+                if (Host.software_version.ContainsKey("date"))
+                {
+                    if (!Util.TryParseIso8601DateTime(Host.software_version["date"], out softwareVersionDate))
+                        Util.TryParseNonIso8601DateTime(Host.software_version["date"], out softwareVersionDate);
+                }
+
+                if (Host.last_software_update > softwareVersionDate && Host.last_software_update > unixMinDateTime)
+                {
+                    lastUpdateTime = HelpersGUI.DateTimeToString(Host.last_software_update.ToLocalTime(), Messages.DATEFORMAT_DMY_HMS, true);

```

Comment:
Can't we try to use the `Pool.last_update_sync.ToLocalTime()` from the host's pool?

If it's not possible then let's leave it empty but I can go to the host's General Tab and see last sync there, and then it doesn't show here.

Reply:
I understand we might not want to show it for every row in a pool with several hosts. So maybe only show if the host is standalone?